### ISSUE TITLE
Prelude update response errored& TLS initialization failed??

### ISSUE THREAD

**Issue by nightblue2536** [loot/loot#1720]
![Screenshot 2022-03-13 112008](https://user-images.githubusercontent.com/101540854/158105307-b32f69d0-b6c2-47b0-9ade-ba34364534d3.jpg)
 hi I'm horrible  time trying figure it out what this is I just updated the new loot it kept saying update the masterlist in which I tried too but no this kept popping up and now I can't do anything right now it saying "Failed to check for LOOT updates! you can check your LOOTDebugLog.txt ( you can get to it through the main menu) for more information. I have no idea what is that can you tell more about please? I was used 17 version until the new update.

**Comment by sibir-ine** [comment]
@nightblue2536, please see [this Q/A](https://loot.github.io/docs/help/LOOT-FAQs.html#tls-failure) from the FAQ.

**Comment by nightblue2536** [comment]
> @nightblue2536, please see [this Q/A](https://loot.github.io/docs/help/LOOT-FAQs.html#tls-failure) from the FAQ.

I fixed the problem thank you the Q/A doesn't not help for me but I figure it out

**Comment by sibir-ine** [comment]
What did you have to do to resolve the issue?

**Comment by Ortham** [comment]
As the FAQ entry suggests, we've seen this error reported by people trying to use the 32-bit build on a 64-bit OS, and switching to the 64-bit build fixed it for them. However, I'm able to run the 32-bit build without issues, so I'd be interested in finding out what OS versions people who experience this issue are running to see if there's some common factor.

**Comment by cooly1975** [comment]
i have exact the same problem. when it is because of the 32 bit version on a 64 bit system, then the question is where i get the 64 bit version for win7 64? the actuel build is for win 10 or higher.

**Comment by sibir-ine** [comment]
> i have exact the same problem. when it is because of the 32 bit version on a 64 bit system, then the question is where i get the 64 bit version for win7 64? the actuel build is for win 10 or higher.

There isn't one at the moment as the 32-bit build is intended to avoid dropping support for both users on Windows 7 & on 32-bit systems. Ideally, we only want to have to provide one additional build to continue to support both users if possible.

Can you please try following the advice from the second paragraph of this [Q/A](https://loot.github.io/docs/help/LOOT-FAQs.html#update-masterlist-errors)?

> If you're on Windows 7, this may be because on the 22nd of February 2018, GitHub disabled support for connections using a version of TLS older than 1.2, and Windows 7 doesn't support TLS 1.2 by default. Windows 7 users must ensure they have the latest Windows updates, and that they have [enabled TLS 1.2 support](https://support.microsoft.com/en-us/help/3140245/update-to-enable-tls-1-1-and-tls-1-2-as-a-default-secure-protocols-in).

If it doesn't work or isn't applicable, can you please turn on debug logging & provide a full debug log (see [this guide](https://loot.github.io/docs/help/How-To-Report-Issues-Helpfully.html#supplying-log) for instructions)?

**Comment by cooly1975** [comment]
i did install the fils manuel with wget for windows can be found here "https://eternallybored.org/misc/wget/" the i downloaded the 2 files prelude and masterlist manuel and copy it to the app folder. now its working

**Comment by Ketrel** [comment]
I'm in the same boat on one of my gaming machines.  That patch is installed, and the registry keys do exist.  Older versions of LOOT such as 0.12.5 (which I had hanging around) still work.

I saved the debug log (blanked out a name, replaced with ___NAME___).  That's available here: https://gist.github.com/Ketrel/88dea26a8daf68afe7b98bc9f96a1571

**Comment by Ortham** [comment]
> I'm in the same boat on one of my gaming machines. That patch is installed, and the registry keys do exist. Older versions of LOOT such as 0.12.5 (which I had hanging around) still work.
> 
> I saved the debug log (blanked out a name, replaced with __NAME__). That's available here: https://gist.github.com/Ketrel/88dea26a8daf68afe7b98bc9f96a1571

Can you try [this build](https://loot.jfrog.io/ui/native/loot/master/) and upload the log file it gives to this issue?

**Comment by Ortham** [comment]
I'm able to replicate this issue in the "IE11 on Win7 (x86)" HyperV VM downloaded from [here](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/), once I finish running through all the Windows Update updates available I'll see if it still occurs and try to figure out how to fix it.

**Comment by Ortham** [comment]
TL;DR: Try installing the [MSVC 2010 x86 redistributable](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2010-vc-100-sp1-no-longer-supported).

After installing all available Windows updates (apart from language packs), starting LOOT would display an error dialog saying

> The program can't start befcause MSVCR100.dll is missing from your computer. Try reinstalling the program to fix this problem.

However, LOOT would actually run and displayed an error saying that the LOOT update check failed. The debug log contained the same error code 99 TLS initialization failed error that others have reported.

I downloaded the MSVC 2010 x86 redistributable from [here](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2010-vc-100-sp1-no-longer-supported), installed it and then re-ran LOOT. This time there was no error dialog, and LOOT's update check didn't error. I then ran masterlist update and that worked fine too. There are also no update errors in LOOT's debug log.

The MSVC 2010 redist requirement is a surprise to me, so I did a bit of digging: Dependency Walker shows that it's pulled in by `libcrypto-1_1.dll` and `libssl-1_1.dll`, which are only used by Qt5 builds. I'll have to update the release notes to mention the dependency and also update the installer.

I also used Process Explorer and noticed that the Qt5 build doesn't actually load `D3Dcompiler_47.dll`, `libEGL.dll`, `libGLESv2.dll` or `opengl32sw.dll`, and I then checked the Qt6 build and it doesn't load `D3Dcompiler_47.dll` or `opengl32sw.dll`, so I think it's safe to leave them out when packaging LOOT, which will save some more space.

**Comment by Ortham** [comment]
I think this is fixed as of 7b9d26072dbb670162f33d820f99613e2db8984b, but nobody else mentioned seeing an error mentioning `MSVCR100.dll`, so could those experiencing this issue please install the MSVC 2010 x86 redistributable (linked above) and let me know if that resolves the issue?

**Comment by Ketrel** [comment]
I'll be at the place with the Win 7 machine to check.  But keep in mind these are 64bit Win 7 installs, and the MSVC 2010 runtime (x86 and x64) are 100% present on the machine.  I can look and verify, but I know it is.  

For the specific build you wanted the log from, is it still the one at that link? It seems to be showing the current master rather than any specific commit's build.

**Comment by Ortham** [comment]
> For the specific build you wanted the log from, is it still the one at that link? It seems to be showing the current master rather than any specific commit's build.

Running the latest master build is fine.

**Comment by Ortham** [comment]
The installer doesn't have the OpenSSL DLLs, that's probably why it's not working. 🤦

**Comment by Ortham** [comment]
The latest master branch build's installer now includes those DLLs, so LOOT installed using the 32-bit installer should work now.

**Comment by Ortham** [comment]
Done in 8fb92eb2f2e1eeccf101acba25410811d1087034, closing this.
